packages/muelu/test/scaling/MMKernelDriver.cpp

```diff
@@ -403,107 +403,122 @@ void Multiply_KokkosKernels(const Xpetra::Matrix<Scalar,LocalOrdinal,GlobalOrdin
+      typedef Tpetra::CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node> crs_matrix_type;
+      typedef Tpetra::Import<LocalOrdinal,GlobalOrdinal,Node>          import_type;
+      typedef typename crs_matrix_type::local_matrix_type    KCRS;
+      typedef typename KCRS::device_type device_t;
+      typedef typename KCRS::StaticCrsGraphType graph_t;
+      typedef typename graph_t::row_map_type::non_const_type lno_view_t;
+      typedef typename graph_t::entries_type::non_const_type lno_nnz_view_t;
+      typedef typename KCRS::values_type::non_const_type scalar_view_t;
+      typedef Kokkos::View<LO*, typename lno_view_t::array_layout, typename lno_view_t::device_type> lo_view_t;
+      typedef Tpetra::Map<LO,GO,NO>                                     map_type;
+      typedef typename map_type::local_map_type                         local_map_type;
+      typedef typename Node::execution_space execution_space;
+      typedef Kokkos::RangePolicy<execution_space, size_t> range_type;
+      LocalOrdinal LO_INVALID = Teuchos::OrdinalTraits<LO>::invalid();
+      RCP<const import_type> Cimport;
+      RCP<const crs_matrix_type> Au = Utilities::Op2TpetraCrs(rcp(&A,false));
+      RCP<const crs_matrix_type> Bu = Utilities::Op2TpetraCrs(rcp(&B,false));
+      RCP<const crs_matrix_type> Cu = Utilities::Op2TpetraCrs(rcp(&C,false));
+      RCP<crs_matrix_type> Cnc = Teuchos::rcp_const_cast<crs_matrix_type>(Cu);
+
+      //    if(!Au->getComm()->getRank())
+      //      std::cout<< "Kokkos::Compat::KokkosOpenMPWrapperNode::execution_space::concurrency() = "<<Kokkos::Compat::KokkosOpenMPWrapperNode::execution_space::concurrency()<<std::endl;
+
+
+
+      // **********************************
+      // Copy in the data for LTG
+      tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("MM LTG: CopyIn")));
+
+      Tpetra::CrsMatrixStruct<Scalar, LocalOrdinal, GlobalOrdinal, Node> Aview, Bview;
+      Aview.origMatrix   = Au;
+      Aview.origRowMap   = Au->getRowMap();
+      Aview.rowMap       = Au->getRowMap();
+      Aview.colMap       = Au->getColMap();
+      Aview.domainMap    = Au->getDomainMap();
+      Aview.importColMap = Teuchos::null;
+
+      Bview.origMatrix   = Bu;
+      Bview.origRowMap   = Bu->getRowMap();
+      Bview.rowMap       = Bu->getRowMap();
+      Bview.colMap       = Bu->getColMap();
+      Bview.domainMap    = Bu->getDomainMap();
+      Bview.importColMap = Teuchos::null;
+
+
+      // Because we're in serial...
+      Cnc->replaceColMap(Bu->getColMap());
+
+      // Bcol2Ccol is trivial
+      lo_view_t Bcol2Ccol(Kokkos::ViewAllocateWithoutInitializing("Bcol2Ccol"),Bview.colMap->getNodeNumElements()), Icol2Ccol;
+      const LO colMapSize = static_cast<LO>(Bview.colMap->getNodeNumElements());
+      Kokkos::parallel_for("Tpetra::mult_A_B_newmatrix::Bcol2Ccol_fill",
+                           Kokkos::RangePolicy<execution_space, LO>(0, colMapSize),
+                           KOKKOS_LAMBDA(const LO i) {
```

Comment:
Is this really supposed to be an OpenMP-only specialization?  `KOKKOS_LAMBDA` marks the lambda as a CUDA device function.

Reply:
@cgcgcg thought low-thread Gustavson SPGEMM would be too slow on CUDA, but nobody has actually tried it yet. I was planning to see if it actually is slow. If it's not slow I would just make this kernel the default.

Reply:
(I would do this as a separate, later PR)